File: src/main/id2h/yatm/creativetabs/CreativeTabsYATM.scala
package id2h.yatm.creativetabs

import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly
import id2h.yatm.Blocks
import net.minecraft.item.Item
import net.minecraft.creativetab.CreativeTabs

class CreativeTabsYATM(name: String) extends CreativeTabs(name) {
	@SideOnly(Side.CLIENT)
	override def getTabIconItem(): Item = Item.getItemFromBlock(Blocks.chassis)
}

object CreativeTabsYATM {
	private val INSTANCE: CreativeTabs = new CreativeTabsYATM("yatm")

	def instance() = INSTANCE
}

File: src/main/id2h/yatm/creativetab/CreativeTabsYATM.java
package id2h.yatm.creativetab;

import id2h.yatm.init.Blocks;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

import net.minecraft.item.Item;
import net.minecraft.creativetab.CreativeTabs;

public class CreativeTabsYATM extends CreativeTabs
{
	private static final CreativeTabs INSTANCE = new CreativeTabsYATM();

	public CreativeTabsYATM()
	{
		super("yatm");
	}

	@Override
	@SideOnly(Side.CLIENT)
	public Item getTabIconItem()
	{
		return Item.getItemFromBlock(Blocks.chassis);
	}

	public static CreativeTabs instance()
	{
		return INSTANCE;
	}
}
